packages/dts-test/defineComponent.test-d.tsx

```diff
@@ -337,26 +337,28 @@ describe('with object props', () => {
   })
 })
 
-describe('type inference w/ optional props declaration', () => {
-  const MyComponent = defineComponent<{ a: string[]; msg: string }>({
-    setup(props) {
-      expectType<string>(props.msg)
-      expectType<string[]>(props.a)
-      return {
-        b: 1,
-      }
-    },
-  })
-
-  expectType<JSX.Element>(<MyComponent msg="1" a={['1']} />)
-  // @ts-expect-error
-  ;<MyComponent />
-  // @ts-expect-error
-  ;<MyComponent msg="1" />
-})
+// describe('type inference w/ optional props declaration', () => {
+//   const MyComponent = defineComponent<{ a: string[]; msg: string }>({
+//     setup(props) {
+//       expectType<string>(props.msg)
+//       expectType<string[]>(props.a)
+//       return {
+//         b: 1,
+//       }
+//     },
+//   })
+
+//   expectType<JSX.Element>(<MyComponent msg="1" a={['1']} />)
+//   // @ts-expect-error
+//   ;<MyComponent />
+//   // @ts-expect-error
+//   ;<MyComponent msg="1" />
+// })
```

Comment:
Commented because I don't think this is correct, because the `props` as type will not be accessible as `props` but in the `attrs`, to override component props,  we can recommend the newly added `DeclareComponent` helper